### PR TITLE
Blog - How the Docker Image Is Stored on the Host Machine

### PR DESCRIPTION
**Pull request by msali123**

### CODE REVIEW

**Comment by msali123** [comment]
@balapriyac 

[Preview Link](https://642f3387ae0432063a6df2c0--ephemeral-narwhal-3ac3b2.netlify.app/blog/docker-image-storage-on-host/)

Also please have a look at **ChainID Calculation** if it content looks right?

**Comment by balapriyac** [comment]
> @balapriyac
> 
> [Preview Link](https://642f3387ae0432063a6df2c0--ephemeral-narwhal-3ac3b2.netlify.app/blog/docker-image-storage-on-host/)
> 
> Also please have a look at **ChainID Calculation** if it content looks right?

Yeah, the ChainID part is correct in terms of the content. The ID is of length 64 so it'll anyways extend to the next line. I'll suggest some formatting edits. Let's see if it makes it look better.

File: blog/_posts/2023-05-31-docker-image-storage-on-host.md [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+]
+~~~
+
+### ChainID Calculation
+
+![Calculation]({{site.images}}{{page.slug}}/calculation.png)\
+
+The ChainID is calculated by concatenating the DiffIDs of the layers in the image and hashing the result using the SHA256 algorithm. It is also used to verify the integrity of an image during transmission. Using the SHA256 values for the current and previous layers to determine the chainID. The formula is shown below.
+
+DiffID = chainID if the layer is the lowest layer among other layers.
```
Comment (balapriyac) [inline blog/_posts/2023-05-31-docker-image-storage-on-host.md:299]:
```suggestion
**DiffID = ChainID** if the layer is the lowest layer among other layers.
```

File: blog/_posts/2023-05-31-docker-image-storage-on-host.md [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+
+If not, chainID(n) = sha256sum [diffID(n-1), diffID(n)]
```
Comment (balapriyac) [inline blog/_posts/2023-05-31-docker-image-storage-on-host.md:301]:
```suggestion
If not, **ChainID(n) = sha256sum [DiffID(n-1), DiffID(n)]**
```

File: blog/_posts/2023-05-31-docker-image-storage-on-host.md [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+
+**Let's calculate the ChainID for Layer-1:**
```
Comment (balapriyac) [inline blog/_posts/2023-05-31-docker-image-storage-on-host.md:303]:
```suggestion
**Let's calculate the ChainID for Layer 1:**
```

File: blog/_posts/2023-05-31-docker-image-storage-on-host.md [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+ChainID(Layer-1) = diffID(Layer-1)
```
Comment (balapriyac) [inline blog/_posts/2023-05-31-docker-image-storage-on-host.md:305]:
```suggestion
ChainID(Layer 1) = DiffID(Layer 1)
```

File: blog/_posts/2023-05-31-docker-image-storage-on-host.md [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+
+ChainID(Layer-1) = d3cc7b6aa7bc15725c1a856ce06fe436da3fbccf0c9c06b04e45f79b3439c154
```
Comment (balapriyac) [inline blog/_posts/2023-05-31-docker-image-storage-on-host.md:307]:
```suggestion
ChainID(Layer 1) = d3cc7b6aa7bc15725c1a856ce06fe436da3fbccf0c9c06b04e45f79b3439c154
```

File: blog/_posts/2023-05-31-docker-image-storage-on-host.md [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+
+**Let's calculate the ChainID for Layer-2:**
+
+ChainID(Layer-2) = sha256sum [diffID(Layer-1), diffID(Layer-2)
```
Comment (balapriyac) [inline blog/_posts/2023-05-31-docker-image-storage-on-host.md:311]:
```suggestion
ChainID(Layer 2) = sha256sum [DiffID(Layer 1), DiffID(Layer 2)
```

File: blog/_posts/2023-05-31-docker-image-storage-on-host.md [hunk 7]
```diff
@@ -0,0 +1,392 @@
...
+
+diffID(Layer-1) = d3cc7b6aa7bc15725c1a856ce06fe436da3fbccf0c9c06b04e45f79b3439c154
```
Comment (balapriyac) [inline blog/_posts/2023-05-31-docker-image-storage-on-host.md:313]:
```suggestion
DiffID(Layer 1) = d3cc7b6aa7bc15725c1a856ce06fe436da3fbccf0c9c06b04e45f79b3439c154
```

File: blog/_posts/2023-05-31-docker-image-storage-on-host.md [hunk 8]
```diff
@@ -0,0 +1,392 @@
...
+
+diffID(Layer-2) = a7f421510691bf6a7b344d1efb738b3d343e252e7dde114a0dd86d432ef6000c
```
Comment (balapriyac) [inline blog/_posts/2023-05-31-docker-image-storage-on-host.md:315]:
```suggestion
DiffID(Layer 2) = a7f421510691bf6a7b344d1efb738b3d343e252e7dde114a0dd86d432ef6000c
```

File: blog/_posts/2023-05-31-docker-image-storage-on-host.md [hunk 9]
```diff
@@ -0,0 +1,392 @@
...
+diffID(Layer-3) = 6ac2db160c6cf3dcf1aff0ced069aa98da28c50cff5cd3c8881c04f42e3ef1fe
+
+~~~{.bash caption=">_"}
+
+$ echo -n "sha256:a7f421510691bf6a7b344d1efb738b3d343e252e7dde114a0dd86d432ef6000c sha256:6ac2db160c6cf3dcf1aff0ced069aa98da28c50cff5cd3c8881c04f42e3ef1fe" | sha256sum
+
+9b3abbf0ab6402c9bcb9cce411268ffe24573d790f0333d8ae06794313295dbd
+~~~
+
+ChainID(Layer-3) = 9b3abbf0ab6402c9bcb9cce411268ffe24573d790f0333d8ae06794313295dbd
```
Comment (balapriyac) [inline blog/_posts/2023-05-31-docker-image-storage-on-host.md:340]:
```suggestion
ChainID(Layer 3) = 9b3abbf0ab6402c9bcb9cce411268ffe24573d790f0333d8ae06794313295dbd
```

File: blog/_posts/2023-05-31-docker-image-storage-on-host.md [hunk 10]
```diff
@@ -0,0 +1,392 @@
...
+
+ChainID(Layer-2) = 29bd3d7c6e1683e422776b9d3285e8a3f1272f07656fc63a941cb7729a169100
+
+Let's calculate the ChainID for Layer-3:
+
+ChainID(Layer-3) = sha256sum [diffID(Layer-2), diffID(Layer-3)
+
+diffID(Layer-2) = a7f421510691bf6a7b344d1efb738b3d343e252e7dde114a0dd86d432ef6000c
+diffID(Layer-3) = 6ac2db160c6cf3dcf1aff0ced069aa98da28c50cff5cd3c8881c04f42e3ef1fe
+
```
Comment (balapriyac) [inline blog/_posts/2023-05-31-docker-image-storage-on-host.md:332]:
```suggestion
~~~

ChainID(Layer 2) = 29bd3d7c6e1683e422776b9d3285e8a3f1272f07656fc63a941cb7729a169100

**Next, let's calculate the ChainID for Layer 3:**

ChainID(Layer 3) = sha256sum [DiffID(Layer 2), DiffID(Layer 3)

DiffID(Layer 2) = a7f421510691bf6a7b344d1efb738b3d343e252e7dde114a0dd86d432ef6000c
DiffID(Layer 3) = 6ac2db160c6cf3dcf1aff0ced069aa98da28c50cff5cd3c8881c04f42e3ef1fe

```

**Review by balapriyac — CHANGES_REQUESTED** [review]
Looks great. Suggested a few formatting edits.

**Comment by msali123** [comment]
@balapriyac 
[Changes done](https://642ff8dfb8a2f621e7d34dd5--ephemeral-narwhal-3ac3b2.netlify.app/blog/docker-image-storage-on-host/)

**Comment by balapriyac** [comment]
@jalletto this is ready to merge.

**Review by jalletto — APPROVED** [review]

**Pull request merged**